sympy/physics/continuum_mechanics/beam.py

```diff
@@ -70,29 +71,42 @@ def __init__(self, length, elastic_modulus, second_moment, variable=Symbol('x'),
 
         Parameters
         ==========
+
         length : Sympifyable
             A Symbol or value representing the Beam's length.
+
         elastic_modulus : Sympifyable
             A SymPy expression representing the Beam's Modulus of Elasticity.
             It is a measure of the stiffness of the Beam material. It can
             also be a continuous function of position along the beam.
-        second_moment : Sympifyable
-            A SymPy expression representing the Beam's Second moment of area.
-            It is a geometrical property of an area which reflects how its
-            points are distributed with respect to its neutral axis. It can
-            also be a continuous function of position along the beam.
+
+        second_moment : Sympifyable or Geometry object
+            Describes the cross-section of the beam  via a sympy expression
+            representing the Beam's Second moment of area. It is a geometrical
+            property of an area which reflects how its points are distributed
+            with respect to its neutral axis. It can also be a continuous
+            function of position along the beam. Alternatively ``second_moment``
+            can be a geometry object such as a ``Polygon`` from the geometry module
+            representing the shape of the cross-section of the beam. The second moment
+            of area will be computed from the gemetry object internally.
```

Comment:
```suggestion
            of area will be computed from the Polygon internally.
```

Reply:
But ellipses, and circles are also being considered, so is it okay to specify 'Polygon' here?

Reply:
Polygon is too specific but there isn't a good term in the mro to describe what you want to accept:
```
In [22]: Polygon.mro()                                                                                                            
Out[22]: 
[sympy.geometry.polygon.Polygon,
 sympy.geometry.entity.GeometrySet,
 sympy.geometry.entity.GeometryEntity,
 sympy.sets.sets.Set,
 sympy.core.basic.Basic,
 object]
```
I would just use "shape" in the docstring. In any case "gemetry" is missing an "o".